gordo_components/builder/build_model.py

```diff
@@ -85,34 +90,48 @@ def build_model(
     cv_duration_sec = None
 
     if cv_mode.lower() in ("cross_val_only", "full_build"):
-
+        metrics_list = [
+            explained_variance_score,
+            r2_score,
+            mean_squared_error,
+            mean_absolute_error,
+        ]
         # Cross validate
         logger.debug("Starting cross validation")
         start = time.time()
-
-        scores: Dict[str, Any]
-
+        scores: Dict[str, Any] = dict()
         if hasattr(model, "predict"):
 
-            cv_scores = cross_val_score(
+            metrics_dict = get_metrics_dict(metrics_list, y)
+
+            n_splits = 3
+            cv = cross_validate(
                 model,
                 X,
                 y,
-                scoring=make_scorer(metric_wrapper(explained_variance_score)),
-                cv=TimeSeriesSplit(n_splits=3),
+                scoring=metrics_dict,
+                return_estimator=True,
+                cv=TimeSeriesSplit(n_splits=n_splits),
             )
 
-            scores = {
-                "explained-variance": {
-                    "mean": cv_scores.mean(),
-                    "std": cv_scores.std(),
-                    "max": cv_scores.max(),
-                    "min": cv_scores.min(),
-                    "raw-scores": cv_scores.tolist(),
+            for md in metrics_dict.items():
+                key = md[0]
+                val = {
+                    "fold-mean": cv[f"test_{key}"].mean(),
+                    "fold-std": cv[f"test_{key}"].std(),
+                    "fold-max": cv[f"test_{key}"].max(),
+                    "fold-min": cv[f"test_{key}"].min(),
                 }
-            }
+                val.update(
+                    {
+                        f"fold-{i + 1}": raw_value
+                        for i, raw_value in enumerate(cv[f"test_{key}"].tolist())
```

Comment:
instead of all these `f"test_{key}"` , maybe create a variable, or update `key` a map during the for loop. ie.
```python
for key in map(lambda k: f"test_{k}", metrics_dict):
```